src/Lib.hs

```diff
@@ -70,65 +73,85 @@ collectEmails = do
     let ticketIds = foldr (\TicketInfo{..} acc -> ticketId : acc) [] tickets
     mapM_ extractEmailAddress ticketIds
 
-processTicket :: TicketId -> App ()
+
+processTicket :: TicketId -> App [ZendeskResponse]
 processTicket ticketId = do
+
     -- We first fetch the function from the configuration
-    getTicketInfo <- asksZendeskLayer zlGetTicketInfo
-    putTextLn "Processing single ticket"
-    ticketInfo <- getTicketInfo ticketId
-    processTicketAndId ticketInfo
-    putTextLn "Process finished, please see the following url"
-    putTextLn $ "https://iohk.zendesk.com/agent/tickets/" <> show ticketId
+    getTicketInfo       <- asksZendeskLayer zlGetTicketInfo
+    printText           <- asksIOLayer iolPrintText
+
+    printText "Processing single ticket"
+
+    ticketInfo          <- getTicketInfo ticketId
+    attachments         <- getTicketAttachments ticketInfo
+
+    zendeskResponse     <- mapM (inspectAttachment ticketInfo) attachments
+
+    postTicketComment   <- asksZendeskLayer zlPostTicketComment
+    _                   <- mapM postTicketComment zendeskResponse
+
+    printText "Process finished, please see the following url"
+    printText $ "https://iohk.zendesk.com/agent/tickets/" <> show ticketId
+
+    pure zendeskResponse
+
 
 processTickets :: App ()
 processTickets = do
-    cfg <- ask
-    sortedTicketIds <- processBatchTickets cfg
-    mapM_ processTicketAndId sortedTicketIds
+    sortedTicketIds     <- listAndSortTickets
+
+    zendeskResponses    <- forM sortedTicketIds $ \ticketInfo -> do
```

Comment:
I think this will cause an issue.
This function first analyzes all the tickets the Zendesk, then start posting comments. Analysis will take a long time so if the process was killed (or failed), it will have to do everything all over again.

Reply:
If a crash occurs, it's not going to be good in any case. In other cases, this is ok. But the plan was to switch to `processTicket` anyway, so I'll do that.